### PR TITLE
Added experimental support for nightly build of Roslyn.

### PR DESCRIPTION
**Pull request by patriksvensson**
To use the "latest" Roslyn bits (version is currently hard coded), pass the script parameter `experimental` to the Cake.exe.

```
C:> Cake.exe build.cake -experimental
```

### CODE REVIEW

File: src/Cake/Scripting/Roslyn/Installation/RoslynInstallerInstructions.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        public IReadOnlyList<FilePath> Paths
+        {
+            get { return _paths; }
+        }
+
+        /// <summary>
+        /// Gets the packages to be installed.
+        /// </summary>
+        /// <value>The packages.</value>
+        public IReadOnlyList<Tuple<string, SemanticVersion>> Packages
```
Comment (devlead) [inline src/Cake/Scripting/Roslyn/Installation/RoslynInstallerInstructions.cs]: `Tuple` needed? Perhaps more readable with an `KeyVauePair<string, SemanticVersion>` ?

Comment (patriksvensson) [inline src/Cake/Scripting/Roslyn/Installation/RoslynInstallerInstructions.cs]: Actually, A `Dictionary<string, SemanticVersion>` would be even better. Think I need some sleep... :smile:

Comment (devlead) [inline src/Cake/Scripting/Roslyn/Installation/RoslynInstallerInstructions.cs]: True go for an `IDictionary<string, SemanticVersion> `;)

**Comment by patriksvensson** [comment]
This is OK to be merged. Should I rebase?

**Comment by devlead** [comment]
Either you or I rebase;) I can take a look and merge on Sunday @patriksvensson :+1:

**Comment by devlead** [comment]
Merged epic PR! Rebased, added usage and fixed a few trailing white spaces.

**Pull request merged**